safenode/src/protocol/types/wallet/local_store.rs

```diff
@@ -0,0 +1,344 @@
+
+        assert_eq!(GENESIS_DBC_AMOUNT, local_wallet.balance().as_nano());
+
+        Ok(())
+    }
+
+    #[test]
+    #[allow(clippy::result_large_err)]
+    fn deposit_does_not_add_dbcs_not_belonging_to_the_wallet() -> GenesisResult<()> {
+        let genesis_key = MainKey::random();
+        let genesis = create_genesis_dbc(&genesis_key)?;
+
+        let wallet_key = MainKey::random();
+        let mut local_wallet: LocalDepositWallet = DepositWallet::new(wallet_key);
+
+        local_wallet.deposit(vec![genesis]);
+
+        assert_eq!(Token::zero(), local_wallet.balance());
+
+        Ok(())
+    }
+
+    /// --------------------------------
+    /// <-------> SendWallet <--------->
+    /// --------------------------------
+
+    #[test]
+    #[allow(clippy::result_large_err)]
+    fn send_wallet_basics() -> GenesisResult<()> {
+        let main_key = MainKey::random();
+        let public_address = main_key.public_address();
+        let client = MockSendClient::new();
+        let send_wallet: LocalSendWallet<MockSendClient> =
+            SendWallet::<MockSendClient>::new(main_key, client);
+
+        assert_eq!(public_address, send_wallet.address());
+        assert_eq!(public_address, send_wallet.new_dbc_address().public_address);
+        assert_eq!(Token::zero(), send_wallet.balance());
+
+        assert!(send_wallet.wallet.available_dbcs.is_empty());
+        assert!(send_wallet.wallet.output_history.is_empty());
+        assert!(send_wallet.wallet.spent_dbcs.is_empty());
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    #[allow(clippy::result_large_err)]
+    async fn sending_decreases_balance() -> Result<()> {
+        let sender_main_key = MainKey::random();
+        let sender_dbc =
+            create_genesis_dbc(&sender_main_key).expect("Genesis creation to succeed.");
+
+        let client = MockSendClient::new();
+        let mut send_wallet: LocalSendWallet<MockSendClient> =
+            SendWallet::<MockSendClient>::new(sender_main_key, client);
+
+        send_wallet.deposit(vec![sender_dbc]);
+
+        assert_eq!(GENESIS_DBC_AMOUNT, send_wallet.balance().as_nano());
+
+        // We send to a new address.
+        let recipient_main_key = MainKey::random();
+        let recipient_public_address = recipient_main_key.public_address();
+        let to = vec![(Token::from_nano(100), recipient_public_address)];
+        let created_dbcs = send_wallet.send(to).await?;
+
+        assert_eq!(1, created_dbcs.len());
+        assert_eq!(GENESIS_DBC_AMOUNT - 100, send_wallet.balance().as_nano());
+
+        let recipient_dbc = &created_dbcs[0];
+        assert_eq!(100, recipient_dbc.amount.value());
+        assert_eq!(
+            &recipient_public_address,
+            recipient_dbc.dbc.public_address()
+        );
+
+        Ok(())
+    }
+
+    #[derive(Clone)]
+    struct MockSendClient {}
+
+    impl MockSendClient {
+        fn new() -> Self {
+            Self {}
+        }
+    }
```

Comment:
Perhaps use a unit struct?

```suggestion
    struct MockSendClient;
```

Reply:
Sure :+1: